src/components/mobile-menu-levels/level-components/index.tsx

```diff
@@ -0,0 +1,79 @@
+// Components
+import {
+	SidebarHorizontalRule,
+	SidebarNavMenuItem,
+} from 'components/sidebar/components'
+// Utils
+import { generateTopLevelSubNavItems } from 'lib/generate-top-level-sub-nav-items'
+import { generateProductLandingSidebarMenuItems } from 'components/sidebar/helpers/generate-product-landing-nav-items'
+import { generateResourcesNavItems } from 'components/sidebar/helpers'
+// Types
+import type { MenuItem } from 'components/sidebar'
+import type { ProductData, ProductSlug } from 'types/products'
+import type { MobileMenuLevelData } from '../types'
+// Styles
+import s from './level-components.module.css'
+
+/**
+ * Render a list of resource nav items for a given product slug.
+ */
+export function ProductResourceNavItems({ slug }: { slug: ProductSlug }) {
+	return (
+		<ul className={s.listResetStyles}>
+			{generateResourcesNavItems(slug).map((item: MenuItem, index: number) => (
+				// eslint-disable-next-line react/no-array-index-key
+				<SidebarNavMenuItem item={item} key={index} />
+			))}
+		</ul>
+	)
+}
+
+/**
+ * Return mobile menu level data for the shared `main` menu pane.
+ */
+export function mobileMenuLevelMain(): MobileMenuLevelData {
+	return {
+		levelButtonText: 'Main Menu',
+		content: (
+			<>
+				<h3 className={s.heading}>Main Menu</h3>
+				<ul className={s.listResetStyles}>
+					{generateTopLevelSubNavItems().map(
+						(item: MenuItem, index: number) => (
+							// eslint-disable-next-line react/no-array-index-key
+							<SidebarNavMenuItem item={item} key={index} />
+						)
+					)}
+				</ul>
+			</>
+		),
+	}
+}
+
+/**
+ * Return mobile menu level data for a common `product` menu pane.
+ */
+export function mobileMenuLevelProduct(
+	productData: ProductData
+): MobileMenuLevelData {
+	return {
+		levelButtonText: `${productData.name} Home`,
+		content: (
+			<ul className={s.listResetStyles}>
+				{[
+					{
+						title: productData.name,
+						fullPath: `/${productData.slug}`,
+						theme: productData.slug,
+					},
+					...generateProductLandingSidebarMenuItems(productData),
```

Comment:
What are your thoughts on this component being a pure 'view' and only rendering the sidebar data as opposed to also generating it?

Reply:
I'd be very on board! The initial intent here with `mobile-menu-level` and especially `level-components` was to make it _possible_ to not use "sidebar-nav-data` context to generate multi-level mobile menus, while keeping things consistent with existing mobile menus.

I made a task 🎟 [Refactor mobile menu "level-components" to decouple business logic & presentation](https://app.asana.com/0/1202097197789424/1205093670087688/f) and added some comments to this file to get that context written down! 😅 

In the short term, I took a lil baby step towards splitting out the logic! Let me know if that looks alright for now 🙇